src/subscription/subscription.controller.ts

```diff
@@ -164,12 +163,11 @@ export class SubscriptionController {
     }
   }
 
-  @Get('/user-subscriptions/:profileId')
-  @IdCheck('profileId')
+  @Get('user-subscriptions')
   @ApiOperation({ summary: 'Get all apis a user is subscribed to' })
-  async getAllSubscriptions(@Param('profileId') profileId: string) {
+  async getAllSubscriptions(@Req() req: Request) {
```

Comment:
By using id from context, you are getting all Apis the same user is subscribed to, that's not the aim here.